### ISSUE TITLE
Allow customizing request model serialization options

### ISSUE THREAD

**Issue by Paillat-dev** [pmateusz/meatie#78]
When using the library with Pydantic v2 models for API clients, there's no way to customize how request models are serialized before being sent. This is particularly problematic with optional fields that default to `None` - they are included in the request JSON even when they weren't explicitly set.

The serialization happens here: https://github.com/pmateusz/meatie/blob/7043da4612ba7fd6b9042463faf999cc8d47d208/src/meatie/internal/adapter/pydantic_v2.py#L37

For example, with this client code:
```python
class MyClient(Client):
    @endpoint("/checkouts", private, method="POST")
    async def _create_checkout(self, request: Annotated[APIRequest, api_ref("body")]) -> APIResponse: ...
```

And this request model with Pydantic v2:
```python
class APIRequest(BaseModel):
    model_config = ConfigDict(ser_json_exclude_unset=True)  # This is ignored
    data: CheckoutRequest

request = APIRequest(
    data=CheckoutRequest(
        type="checkouts",
        attributes=CheckoutRequestAttributes(
            custom_price=50000

        )
    )
)
```

The JSON sent still includes all the None fields despite the model config:
```json
{
    "data": {
        "type": "checkouts",
        "attributes": {
            "custom_price": 50000,
            "product_options": null,
            "checkout_options": null,
            "checkout_data": null,
            "preview": null,
            "test_mode": null,
            "expires_at": null
        }
    }
}
```

## Desired Behavior
We should be able to configure how request models are serialized, particularly the ability to exclude `None` values using something like Pydantic's `exclude_unset=True` or `exclude_none=True`.

The expected request JSON should only include set fields:
```json
{
    "data": {
        "type": "checkouts",
        "attributes": {
            "custom_price": 50000
        }
    }
}
```

## Possible Solution
A solution would be to add serialization options to the `api_ref` annotation:
```python
@endpoint("/checkouts", private, method="POST")
async def _create_checkout(
    self, 
    request: Annotated[APIRequest, api_ref("body", exclude_unset=True)]
) -> APIResponse: ...
```

## Context
This is particularly important when working with APIs that have different behavior for explicitly set `null` values versus omitted fields. Many APIs treat `"field": null` differently from the field being absent in the request.

For example, the LemonSqueezy API uses the absence of fields to indicate default values should be used, while explicit `null` values might override defaults.

Thank you in any case for this truly amazing library, it is very enjoyable to work with.

**Comment by DefiDebauchery** [comment]
FWIW, this was addressed in #70, and I added an example in https://github.com/pmateusz/meatie/issues/70#issuecomment-2424202508

Having said that, directly passing options would certainly be an improvement, as well.

**Comment by Paillat-dev** [comment]
The `unwrap` solution from #70 does not work in my case because it is not ran on `body` fields. To have it work i had to avoid typing my field as anything subclassing Pydantic's BaseModel and use `fmt` instead of `unwrap` (even tough it was inputted as a pydantic model). Avoiding `BadeModel` was needed because else it would try to dump it twice, from my `fmt` and internally. This is breaks type safety and uses parameters in unintended ways.

**Comment by Paillat-dev** [comment]
Specifically, I get this warning:
```
  Expected `APIRequest` but got `dict` with value `{'data': {'type': 'checko...ts', 'id': '593812'}}}}}` - serialized value may not be as expected
  return self.serializer.to_python(
```

**Comment by pmateusz** [comment]
Hi @Paillat-dev,

Thank you for the detailed feature request and including code examples!

I implemented a solution for the improvement you suggested in the PR: https://github.com/pmateusz/meatie/pull/79

The improvement allows using the `fmt` option to customize the serialization of the request body. The `fmt` option accepts a callback that given the request body, should produce either a `str` or `bytes`. The result is sent using aiohttp as `data` (https://docs.aiohttp.org/en/stable/client_reference.html#aiohttp.request).
This way we avoid the second serialization of a JSON payload into a JSON string.

I am proposing this solution because it is the most flexible I found. It gives the user full control over how the request body is serialized.
